Codo-Baggins/refactor_tractor_fitlit
# Refactor - Class Tests First Pass

# Pull Request Template

#### What’s this PR do?
This PR takes a first pass at cleaning up and passing all tests for the `UserRepo`, `User`, `Activity`, `Sleep` and `Hydration` classes. Specifically it:

1. Refactors tests to make most `it` blocks have only have one assertion each.
1. Introduces arrow functions in all tests.
1. Refactors longer methods to limit iterator chaining and makes them more readable.
1. Tests helper methods that were not previously tested.
1. Cleans up spacing discrepancies between methods and `it` blocks


#### Where should the reviewer start?
* Review `User-repo.js` and `user-repo-test.js`
* Review `User.js` and `user-test.js`
* Review `Actvitiy.js` and `Activity-test.js`
* Review `Hyrdation.js` and `Hydration-test.js`
* Review `Sleep.js` and `Sleep-test.js`

#### How should this be manually tested?
* Review how the tests and implementation to see what has been changed from the original.
* Run `npm` test to confirm all the tests are passing.


#### Any background context you want to provide?
* There is more refactoring to be done here, but we will want to get into `inheritance` before we start changig too much more. That will DRY up the code significantly.


#### What are the relevant tickets?
* https://github.com/Codo-Baggins/refactor_tractor_fitlit/issues/2
